exercises/phone-number/phone_number_test.go

```diff
@@ -20,19 +22,24 @@ var numberTests = []testCase{
 }
 
 func TestNumber(t *testing.T) {
+	if testVersion != targetTestVersion {
+		t.Errorf("Found testVersion = %v, want %v.", testVersion, targetTestVersion)
+	}
 	for _, test := range numberTests {
 		actual, actualErr := Number(test.input)
-		if actual != test.expected {
-			t.Errorf("Number(%s): expected [%s], actual: [%s]", test.input, test.expected, actual)
-		}
-		// if we expect an error and there isn't one
-		if test.expectErr && actualErr == nil {
+		if !test.expectErr {
+			if actualErr != nil {
+				// if we don't expect an error and there is one
+				var _ error = actualErr
+				t.Fatalf("Number(%s): expected no error, but error is: %s", test.input, actualErr)
+			}
+			if actual != test.expected {
+				t.Fatalf("Number(%s): expected [%s], actual: [%s]", test.input, test.expected, actual)
+			}
+		} else if actualErr == nil {
+			// if we expect an error and there isn't one
 			t.Errorf("Number(%s): expected an error, but error is nil", test.input)
 		}
```

Comment:
Ah right. What you say is true - however, I believe the `var _ error = actualErr` a few lines above has already constrained the error type. It is OK that it is in a separate conditional branch - the compiler still checks it. I think the following Go code example shows what I mean:

```go
package main

func notAnError() int {
    return 5
}

func main() {
	err := notAnError()
	if false {
		var _ error = err
	}
}
```

This *does* fail to compile, so I think that means we are OK on this front.